pkg/sources/gitlab/gitlab.go

```diff
@@ -315,28 +322,30 @@ func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk)
 			}
 			defer os.RemoveAll(path)
 			if err != nil {
-				errsMut.Lock()
-				errs = append(errs, err)
-				errsMut.Unlock()
-				return
+				scanErrs = append(scanErrs, err)
+				return nil
 			}
-			log.Debugf("Starting to scan repo %d/%d: %s", i+1, len(s.repos), repoURL)
+
+			log.Debugf("Starting to scan repoURL #%d of %d: %s", cnt, repoCnt, repo)
 			err = s.git.ScanRepo(ctx, repo, path, git.NewScanOptions(), chunksChan)
 			if err != nil {
-				errsMut.Lock()
-				errs = append(errs, err)
-				errsMut.Unlock()
-				return
+				scanErrs = append(scanErrs, err)
+				return nil
 			}
-			log.Debugf("Completed scanning repo %d/%d: %s", i+1, len(s.repos), repoURL)
-		}(ctx, repo, i)
+			atomic.AddUint64(&scanned, 1)
+			log.Debugf("scanned %d/%d repos", scanned, repoCnt)
+			log.Debugf("Completed scanning repoURL # %d of %d: %s", cnt, repoCnt, repo)
+
+			return nil
+		})
 	}
-	wg.Wait()
-	if len(errs) == 0 {
-		s.SetProgressComplete(len(s.repos), len(s.repos), fmt.Sprintf("Completed scanning source %s", s.name), "")
+
+	_ = s.jobPool.Wait()
+	if len(scanErrs) == 0 {
+		s.Complete(fmt.Sprintf("Completed scanning source %s", s.name))
 	}
```

Comment:
**Question:** Should a scan be considered complete even if we encountered errors? What would this look like to the customer?

Reply:
For this I would've thought we only consider it complete if len(scanErrs) is 0, which would mean no errors i think?